pkg/features/features.go

```diff
@@ -37,6 +41,12 @@ type Flags struct {
 	fqdnCallbacks map[string]BoolValueOnChangeCallback
 
 	tamperProtection bool
+
+	// beatsAsOtelReceivers indicates if the Agent run Beats as OpenTelemetry receivers like filebeatreceiver etc.
+	beatsAsOtelReceivers bool
+
+	// monitoringWithOtel indicates if the Agent should use OpenTelemetry components for self-monitoring instead of Beats.
+	monitoringWithOtel bool
```

Comment:
I think there is a case where we want to run monitoring with native otel in the future. So we might just want to use a little different name. What about `monitoringWithOtelReceivers`? Just so in the future we want to add `monitoringWithOtel` where it is using native filelog receiver and a native metric receiver.